From efecd1c724a1db9d0f55402ff7a3dbff4b61872a Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 28 Nov 2023 16:59:23 +0100
Subject: [PATCH 0077/1695] drivers: clk: print clock tree with debug trace
 level

Change clk_print_tree() to print the clock tree with debug trace
level instead of info trace level.

Add a test on TRACE_LEVEL >= TRACE_DEBUG in clk_print_tree()
to ensures no unless code is embedded when the trace level is too low.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git e7e6be4318691cdb04adb3a5180fcb0d347ed4a6 drivers: clk: print clock tree with debug trace level]
Change-Id: Ida9de72a127e2a9b55ad5f53a0e7814d9f71ac4a
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from aac2c7167921322237c0d64582376ca01319ab63]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/drivers/clk/clk.c     | 9 +++++----
 core/include/drivers/clk.h | 2 +-
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/core/drivers/clk/clk.c b/core/drivers/clk/clk.c
index 333d91f68..8f5a6a5d2 100644
--- a/core/drivers/clk/clk.c
+++ b/core/drivers/clk/clk.c
@@ -436,7 +436,7 @@ out:
 	if (!msg)
 		snprintf(msg_end - 4, 4, "...");
 
-	IMSG("%s", msg_buf);
+	DMSG("%s", msg_buf);
 }
 
 static void print_tree(void)
@@ -448,7 +448,7 @@ static void print_tree(void)
 
 #ifdef CFG_DRIVERS_CLK_PRINT_TREE
 	if (SLIST_EMPTY(&clock_list)) {
-		IMSG("-- No registered clock");
+		DMSG("-- No registered clock");
 		return;
 	}
 #endif
@@ -481,8 +481,9 @@ static void print_tree(void)
 
 void clk_print_tree(void)
 {
-	if (IS_ENABLED(CFG_DRIVERS_CLK_PRINT_TREE)) {
-		IMSG("Clock tree summary (informative):");
+	if (IS_ENABLED(CFG_DRIVERS_CLK_PRINT_TREE) &&
+	    TRACE_LEVEL >= TRACE_DEBUG) {
+		DMSG("Clock tree summary (informative):");
 		print_tree();
 	}
 }
diff --git a/core/include/drivers/clk.h b/core/include/drivers/clk.h
index 340a79790..20b7d5405 100644
--- a/core/include/drivers/clk.h
+++ b/core/include/drivers/clk.h
@@ -199,7 +199,7 @@ TEE_Result clk_set_parent(struct clk *clk, struct clk *parent);
 TEE_Result clk_get_rates_array(struct clk *clk, size_t start_index,
 			       unsigned long *rates, size_t *nb_elts);
 
-/* Print current clock tree summary on output console (info trace level) */
+/* Print current clock tree summary to output console with debug trace level */
 #ifdef CFG_DRIVERS_CLK
 void clk_print_tree(void);
 #else
-- 
2.47.2

